packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -208,193 +226,187 @@ class SlideshowView extends GalleryComponent {
         avoidIndividualNavigation,
         ignoreScrollPosition,
         isContinuousScrolling,
+        scrollingUpTheGallery,
       });
     }
     this.removeArrowsIfNeeded();
   }
 
+  getNextItemOrGroupToScrollTo(
+    initiator,
+    direction,
+    ignoreScrollPosition,
+    avoidIndividualNavigation,
+    isAutoTrigger
+  ) {
+    let nextIndex;
+    if (initiator === 'nextItem') {
+      if (ignoreScrollPosition) {
+        nextIndex = this.state.activeIndex;
+      } else {
+        if (
+          avoidIndividualNavigation &&
+          !(this.props.styleParams.groupSize > 1)
+        ) {
+          nextIndex = this.getCenteredItemIdxByScroll();
```

Comment:
@yakovhaiilo Can you reuse this line and line 268?

Reply:
This behaviour is duplicated: 
`nextIndex = this.getCenteredGroupIdxByScroll() + direction;`
It is used twice: 
1. `initiator === 'nextGroup`
2. `initiator === 'nextItem && !ignoreScrollPosition && (avoidIndividualNavigation &&
          !(this.props.styleParams.groupSize > 1))`

So you can make one if at the top level of your function that handles that (and `returns`)
Afterwards you deal with the complicated `nextItem` behaviour